src/apps/lwaftr/conf.lua

```diff
@@ -100,109 +104,96 @@ local lwaftr_conf_spec = {
-               ret = str:sub(pos,pos)
-               pos = pos + 1
-            end
-            return ret
-         end,
-         close = function(self) str = nil end
-      }
+function migrate_conf(old)
+   function convert_ipv4(addr)
+      if addr then return util.ipv4_pton(ipv4:ntop(addr)) end
    end
-   local function test(str, expected)
-      if not lib.equal(expected, load_lwaftr_config(string_file(str))) then
-         error('lwaftr conf parse produced unexpected result; string:\n'..str)
-      end
-   end
-   test([[
-            aftr_ipv4_ip = 1.2.3.4
-            aftr_ipv6_ip = 8:9:a:b:c:d:e:f
-            aftr_mac_b4_side = 22:22:22:22:22:22
-            aftr_mac_inet_side = 12:12:12:12:12:12
-            next_hop6_mac = 44:44:44:44:44:44
-            binding_table = "foo-table.txt"
-            hairpinning = false
-            icmpv4_rate_limiter_n_packets=6e3
-            icmpv4_rate_limiter_n_seconds=2
-            icmpv6_rate_limiter_n_packets=6e3
-            icmpv6_rate_limiter_n_seconds=2
-            inet_mac = 68:68:68:68:68:68
-            ipv4_mtu = 1460
-            ipv6_mtu = 1500
-            max_fragments_per_reassembly_packet = 20
-            max_ipv4_reassembly_packets = 100
-            max_ipv6_reassembly_packets = 50
-            policy_icmpv4_incoming = ALLOW
-            policy_icmpv6_incoming = ALLOW
-            policy_icmpv4_outgoing = ALLOW
-            policy_icmpv6_outgoing = ALLOW
-            v4_vlan_tag = 1092 # 0x444
-            v6_vlan_tag = 1638 # 0x666
-            vlan_tagging = true
-        ]],
-      {
-         aftr_ipv4_ip = ipv4:pton('1.2.3.4'),
-         aftr_ipv6_ip = ipv6:pton('8:9:a:b:c:d:e:f'),
-         aftr_mac_b4_side = ethernet:pton("22:22:22:22:22:22"),
-         aftr_mac_inet_side = ethernet:pton("12:12:12:12:12:12"),
-         next_hop6_mac = ethernet:pton("44:44:44:44:44:44"),
-         binding_table = "foo-table.txt",
-         hairpinning = false,
-         icmpv4_rate_limiter_n_packets=6e3,
-         icmpv4_rate_limiter_n_seconds=2,
-         icmpv6_rate_limiter_n_packets=6e3,
-         icmpv6_rate_limiter_n_seconds=2,
-         inet_mac = ethernet:pton("68:68:68:68:68:68"),
-         ipv4_mtu = 1460,
-         ipv6_mtu = 1500,
-         max_fragments_per_reassembly_packet = 20,
-         max_ipv4_reassembly_packets = 100,
-         max_ipv6_reassembly_packets = 50,
-         policy_icmpv4_incoming = policies['ALLOW'],
-         policy_icmpv6_incoming = policies['ALLOW'],
-         policy_icmpv4_outgoing = policies['ALLOW'],
-         policy_icmpv6_outgoing = policies['ALLOW'],
-         v4_vlan_tag = 0x444,
-         v6_vlan_tag = 0x666,
-         vlan_tagging = true
+   local external = {
+      ip = convert_ipv4(old.aftr_ipv4_ip),
+      mac = old.aftr_mac_inet_side,
+      mtu = old.ipv4_mtu,
+      ingress_filter = old.ipv4_ingress_filter,
+      egress_filter = old.ipv4_egress_filter,
+      allow_incoming_icmp = old.policy_icmpv4_incoming == policies.ALLOW,
```

Comment:
These variable names come directly from the yang schema.  However they are inside an "external_interface" container, e.g. you would access them as conf.external_interface_allow_incoming_icmp, so that seems to be clear enough, no?